File: sdk/unity/sdk/Assets/KingstonDynamics/DragonSong/Protocol/Messaging/Type/Heartbeat/HeartbeatPing.cs
using KingstonDynamics.DragonSong.Protocol.Messaging.Data;
using KingstonDynamics.DragonSong.Protocol.Utils;

namespace KingstonDynamics.DragonSong.Protocol.Messaging.Type.Heartbeat
{
    public class HeartbeatPing : Message
    {
        private const MessageType Type = MessageType.HEARTBEAT_PING;

        public HeartbeatPing(string auditId) : base((int)Type, auditId)
        {
            // Empty
        }

        public HeartbeatPing() : this(Keyinator.GenerateGUID())
        {
            // Empty
        }

        public HeartbeatPing(Readinator reader) : base(reader)
        {
            // Empty
        }
    }
}

File: src/protocol/src/main/java/com/kingstondynamics/dragonsong/protocol/message/type/heartbeat/HeartbeatPing.java
package com.kingstondynamics.dragonsong.protocol.message.type.heartbeat;

import com.kingstondynamics.dragonsong.protocol.message.Message;
import com.kingstondynamics.dragonsong.protocol.message.data.MessageType;
import com.kingstondynamics.dragonsong.protocol.utils.Keyinator;
import com.kingstondynamics.dragonsong.protocol.utils.Readinator;

public class HeartbeatPing extends Message {
    
    private final static MessageType type = MessageType.HEARTBEAT_PING;
    
    public HeartbeatPing(String auditId) {
        super(type.getType(), auditId);
    }
    
    public HeartbeatPing() {
        this(Keyinator.generateGUID());
    }
    
    public HeartbeatPing(Readinator reader) {
        super(reader);
    }
}
